evm_arithmetization/src/witness/operation.rs

```diff
@@ -597,6 +599,39 @@ pub(crate) fn generate_swap<F: RichField, T: Transition<F>>(
     Ok(())
 }
 
+pub(crate) fn generate_incr<F: RichField, T: Transition<F>>(
```

Comment:
I'm actually not sure why it's working. For `INCR2-4`, there's no problem, but for `INCR1` we are reading the stack (and writing) at address `stack_len - 1`. There is no guarantee that the current top of the stack has been written in memory, so I'm surprised the reads don't return a wrong value sometimes.